src/main/java/org/tron/common/runtime/vm/program/InternalTransaction.java

```diff
@@ -92,14 +92,22 @@ public InternalTransaction(Transaction trx,  InternalTransaction.TrxType trxType
       this.transferToAddress = Wallet.generateContractAddress(trx);
       this.note = "create";
       this.value = contract.getNewContract().getCallValue();
+      this.data = contract.getNewContract().getBytecode().toByteArray();
+      if(this.data.length == 0){
+        this.data = null;
+      }
     }
-    else if(trxType.equals(TrxType.TRX_CONTRACT_CALL_TYPE)){
+    else if(trxType.equals(TrxType.TRX_CONTRACT_CALL_TYPE)) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Move this "else" on the same line that the previous closing curly brace. ([squid:RightCurlyBraceSameLineAsNextBlockCheck](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3ARightCurlyBraceSameLineAsNextBlockCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1633&issues=AWak_C2fdPPRVdwyhXKM&open=AWak_C2fdPPRVdwyhXKM)

[ISSUE_KEY]: AWak_C2fdPPRVdwyhXKM